pandas/core/arrays/interval.py

```diff
@@ -243,6 +244,20 @@ def _simple_new(
             )
             raise ValueError(msg)
 
+        # For dt64/td64 we want DatetimeArray/TimedeltaArray instead of ndarray
+        from pandas.core.ops.array_ops import maybe_upcast_datetimelike_array
+
+        left = maybe_upcast_datetimelike_array(left)
+        left = extract_array(left, extract_numpy=True)
```

Comment:
Above we are first ensuring that the arrays passed to `_simple_new` are an index, and then we extract the array again. Is this roundtrip to index and back needed?

Reply:
I think we can avoid the roundtrip eventually, will be best accomplished by being stricter in what we pass to _simple_new

Reply:
Sounds good, would be a nice follow-up